app/src/main/java/io/left/meshim/controllers/RightMeshController.java

```diff
@@ -388,5 +404,44 @@ void handleDataDelivery(RightMeshEvent e) {
             unDeliveredMessageIds.remove(deliveryDataId);
             updateInterface();
         }
+        else if(undeliveredPeerUpdateMessage.containsKey(deliveryDataId)){
+            // removing the message from undeliveredPeerUpdate map since we dont need to send it again.
+            undeliveredPeerUpdateMessage.remove(deliveryDataId);
+        }
+    }
+
+    /**
+     * Runs a timer task when there are undelivered peer updates in the que
+     */
+    void setTimerRunningForUndeliveredPackages(){
+        if(!isTimerRunning && !undeliveredPeerUpdateMessage.isEmpty()){
+            isTimerRunning = true;
+            undeliveredPackageTimer.schedule(new TimerTask() {
```

Comment:
This only schedules to run once, correct? It doesn't loop?

Reply:
correct